src/components/Asset/Edit/FormEditService.tsx

```diff
@@ -0,0 +1,125 @@
+import { ReactElement } from 'react'
+import { Field, Form, useFormikContext } from 'formik'
+import Input from '@shared/FormInput'
+import FormActions from './FormActions'
+import { getFieldContent } from '@utils/form'
+import consumerParametersContent from '../../../../content/publish/consumerParameters.json'
+import { Service } from '@oceanprotocol/lib'
+import { ServiceEditForm } from './_types'
+import IconDownload from '@images/download.svg'
+import IconCompute from '@images/compute.svg'
+import FormEditComputeService from './FormEditComputeService'
+import { defaultServiceComputeOptions } from './_constants'
+
+export default function FormEditService({
+  data,
+  chainId,
+  service,
+  accessDetails
+}: {
+  data: FormFieldContent[]
+  chainId: number
+  service: Service
+  accessDetails: AccessDetails
+}): ReactElement {
+  const formUniqueId = service.id // because BoxSelection component is not a Formik component
+  const { values, setFieldValue } = useFormikContext<ServiceEditForm>()
+
+  const accessTypeOptionsTitles = getFieldContent('access', data).options
+
+  const accessTypeOptions = [
+    {
+      name: `access-${formUniqueId}-download`,
+      value: 'access',
+      title: accessTypeOptionsTitles[0],
+      icon: <IconDownload />,
+      // BoxSelection component is not a Formik component
+      // so we need to handle checked state manually.
+      checked: values.access === 'access'
+    },
+    {
+      name: `access-${formUniqueId}-compute`,
+      value: 'compute',
+      title: accessTypeOptionsTitles[1],
+      icon: <IconCompute />,
+      checked: values.access === 'compute'
+    }
+  ]
+
+  return (
+    <Form style={{ margin: 20 }}>
```

Comment:
also here like in formAddService
you can do a 
cont renderField = (filed) => {
<Field.../>
}